crates/lib/src/install.rs

```diff
@@ -1803,6 +1803,54 @@ pub(crate) async fn install_to_disk(mut opts: InstallToDiskOpts) -> Result<()> {
     Ok(())
 }
 
+/// Check if a directory contains only mount points (or is empty), recursively.
+/// Returns true if all entries in the directory tree are either:
+/// - Mount points (on different filesystems)
+/// - Directories that themselves contain only mount points (recursively)
+/// - The lost+found directory
+///
+/// This handles cases like /var containing /var/lib (not a mount) which contains
+/// /var/lib/containers (a mount point).
+#[context("Checking if directory contains only mount points")]
+fn dir_contains_only_mounts(parent_fd: &Dir, dir_name: &str) -> Result<bool> {
```

Comment:
Could have a unit test for this for the trivial cases at least (no inner mounts). Actually testing the mount case needs an integration test.